registrar.go

```diff
@@ -93,7 +93,7 @@ func (r Registry) GetDriverInterfaces() []interface{} {
 func (r Registry) FilterForCompatible(ctx context.Context) Drivers {
 	var wg sync.WaitGroup
 	mutex := &sync.Mutex{}
-	state := make(map[int]*Driver)
+	state := make([]*Driver, len(r.Drivers))
```

Comment:
I think creating a slice with some length and then appending to it will end up creating entries with nil values.
ref: https://goplay.tools/snippet/PZr5lhINoVo

Reply:
yep, the idea here was to move away from using a `map` for this use case see my comment here https://github.com/jacobweinstock/registrar/pull/1#issuecomment-1116216041